src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -69,30 +75,28 @@
 
     private final List<PodTemplate> templates;
     private final String serverUrl;
-    private final String serverCertificate;
+    @CheckForNull
+    private String serverCertificate;
     private String namespace;
     private final String jenkinsUrl;
-    private final String jenkinsTunnel;
-    private final String credentialsId;
+    @CheckForNull
+    private String jenkinsTunnel;
+    @CheckForNull
+    private String credentialsId;
     private final int containerCap;
 
     private transient Kubernetes connection;
 
     @DataBoundConstructor
-    public KubernetesCloud(String name, List<? extends PodTemplate> templates,
-                           String serverUrl, String serverCertificate, String namespace,
-                           String jenkinsUrl,String jenkinsTunnel,
-                           String credentialsId, String containerCapStr, int connectTimeout, int readTimeout) {
+    public KubernetesCloud(String name, List<? extends PodTemplate> templates, String serverUrl, String namespace,
```

Comment:
As long as the plugin hasn't reached 1.0 - and it is still in some early state - we don't bother with binary compatibility.